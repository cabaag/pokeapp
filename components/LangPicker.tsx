import { Icon, Picker } from 'native-base';
import React, { useCallback, useState } from 'react';

const LangPicker: React.FC = () => {
  const [language, setLanguage] = useState('en');

  const handleChangeLang = useCallback((lang) => {
    console.log(lang);
  }, [])

  return (
    <Picker
      iosHeader="Select your language"
      iosIcon={(
        <Icon
          name="arrow-dropdown-circle"
          style={{ color: "#007aff", fontSize: 25 }}
        />
      )}
      mode="dropdown"
      onValueChange={handleChangeLang}
      selectedValue={language}
      style={{ width: undefined }}
    >
      <Picker.Item label="Wallet" value="key0" />
      <Picker.Item label="ATM Card" value="key1" />
      <Picker.Item label="Debit Card" value="key2" />
      <Picker.Item label="Credit Card" value="key3" />
      <Picker.Item label="Net Banking" value="key4" />
    </Picker>
  );
}

export default LangPicker;