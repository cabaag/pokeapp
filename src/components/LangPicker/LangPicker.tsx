import { Icon, Picker } from 'native-base';
import React, { useCallback, useState, memo } from 'react';
import { useTranslation } from 'react-i18next';

const LangPicker: React.FC = () => {
  const { t, i18n } = useTranslation();
  const [language, setLanguage] = useState(i18n.language);

  const handleChangeLang = useCallback((lang) => {
    i18n.changeLanguage(lang);
    setLanguage(lang);
  }, [])

  return (
    <Picker
      iosHeader={t("select_lang")}
      iosIcon={(
        <Icon
          name="arrow-dropdown-circle"
        />
      )}
      mode="dropdown"
      onValueChange={handleChangeLang}
      placeholder={t("select_lang")}
      selectedValue={language}
      

    >
      <Picker.Item label={t("english")} value="en" />
      <Picker.Item label={t("spanish")} value="es" />
    </Picker>
  );
}

export default memo(LangPicker);