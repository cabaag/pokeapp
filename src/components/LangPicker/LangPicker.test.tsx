import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import LangPicker from './LangPicker';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
  language: 'en',
  languages: ['en'],
}));

function setup(): ShallowWrapper {
  return shallow(
    <LangPicker />
  )
}

describe('<LangPicker />', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = setup();
  })

  it('matchs snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});