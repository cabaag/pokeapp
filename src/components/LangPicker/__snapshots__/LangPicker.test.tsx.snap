// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LangPicker /> matchs snapshot 1`] = `undefined`;
